### PR TITLE
Arielsvn/feature/27 dataset download page

### LINKED ISSUE #27: Downloads Page: Start Download and "Download Started" page
## Description
* When user clicks "Download" on downloads page, start download and show them the "Download Started" page that allows them to enter their email when files are ready for download.

## Mockups
screen shot 2018-05-18 at 2 49 52 pm

### LINKED ISSUE #28: Download: "Your dataset is on the way!" page and "Change email" modals
## Description
If user enters their email, they should be shown a confirmation page with the email they entered.

This page also allows them to change the email they entered with a change email modal.

### Confirmation page
screen shot 2018-05-18 at 3 04 16 pm
### Change email modal
screen shot 2018-05-18 at 3 07 02 pm
### Successfully updated
screen shot 2018-05-18 at 3 07 25 pm

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

closes #27 
closes #28 

## Purpose/Implementation Notes

Implemented pages to download datasets, and allow the user to enter an email to receive an alert with the download link.

This PR also introduces a new way of interacting with the history API from the action creators. Allowing us to do `dispatch(push('/new-url'))` in the future. This approach was copied from https://github.com/reactjs/react-router-redux#what-if-i-want-to-issue-navigation-events-via-redux-actions

## Types of changes

* [x] New feature (non-breaking change which adds functionality)

## Functional tests

- Click download on the downloads page. Makes request to the `/dataset` endpoint with the parameter `start: true` which initiates the processing of the file and redirects to the download dataset page
- Add an email on the download dataset page, shows the user feedback and allows to change the email later.
- When the email is modified, a message is displayed for 5 secs

## Screenshots

download btn

### CODE REVIEW

File: src/common/helpers.js [hunk 1]
```diff
@@ -55,3 +55,7 @@ export async function asyncFetch(url, params) {
   if (!response.ok) throw new Error(response.status);
   return await response.json();
 }
+
+export function getAmazonDownloadLinkUrl(s3_bucket, s3_key) {
+  return `https://s3.amazonaws.com/${s3_bucket}/${s3_key}`;
```
Comment (ramenhog) [inline src/common/helpers.js]: I'm guessing this is just a temporary placeholder for now?

File: src/components/Modal/ModalManager.js [hunk 2]
```diff
@@ -16,16 +16,22 @@ export default class ModalManager extends React.Component {
     modalOpen: false
   };
 
+  showModal = () => this.setState({ modalOpen: true });
+
+  hideModal = () => this.setState({ modalOpen: false });
```
Comment (ramenhog) [inline src/components/Modal/ModalManager.js]: 🙌

File: src/containers/DataSet/index.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+              )}
+            </ModalManager>
+
+            <p>If you haven’t recevied it, please check your spam folders.</p>
+          </section>
+          <div className="dataset__way-image">
+            <img src={NextStepsImage} alt="" />
+          </div>
+        </div>
+        {false && (
```
Comment (ramenhog) [inline src/containers/DataSet/index.js]: Hmm, I'm a little confused about this pattern. Won't this never render those lines of HTML?

File: src/common/helpers.js (hunk 1 above)
Comment (arielsvn) [reply src/common/helpers.js]: Depends on how we're going to generate the download link?? I didn't see any field with that information. Instead what the backend is sending is: `s3_bucket` and `s3_key`. I'm guessing the final download url will look like that?

File: src/containers/DataSet/index.js (hunk 3 above)
Comment (arielsvn) [reply src/containers/DataSet/index.js]: That's for the bottom part of the download page:

image

Deepa, told me to hide those for now until those two pages are created. If you want we can remove them from the component

Comment (ramenhog) [reply src/containers/DataSet/index.js]: ahh ok, can you just add a comment?

Comment (arielsvn) [reply src/containers/DataSet/index.js]: sure, good idea :)

**Pull request merged**